packages/acceptance-tests/pkg-tests-core/sources/utils/tests.ts

```diff
@@ -538,12 +527,26 @@ export const startPackageServer = ({type}: { type: keyof typeof packageServerUrl
 
           const {authorization} = req.headers;
           if (authorization != null) {
-            const auth = validAuthorizations.get(authorization);
-            if (!auth) {
+            const user = validAuthorizations.get(authorization);
+            if (!user) {
               sendError(res, 401, `Invalid token`);
               return;
-            } else if (parsedRequest.type === RequestType.Whoami) {
-              parsedRequest.login = auth;
+            }
+
+            const otp = req.headers[`npm-otp`];
+
+            if (user.requiresOtp && user.otp !== otp) {
+              res.writeHead(401, {
+                [`Content-Type`]: `application/json`,
+                [`www-authenticate`]: `OTP`,
+              });
+
+              res.end();
+              return;
+            }
```

Comment:
I put this here, "on the top", as it seems to me that every authenticated request might need to check OTP. If not and this is somehow incorrect, please let me know.